lighthouse-core/gather/drivers/driver.js

```diff
@@ -219,6 +210,28 @@ class Driver {
   }
 
   /**
+   * If our main document URL redirects, we will update options.url accordingly
+   * As such, options.url will always represent the post-redirected URL.
+   * options.initialUrl is the pre-redirect URL that things started with
+   *
+   * Caveat: only works when network recording enabled for a pass
```

Comment:
seems important for gatherers/audits to be able to assume they don't have to do any work to get the actual url of the current page